### PR TITLE
Optimise cases

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by CR
- [x] tests executed by DN

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,9 +8,10 @@ About changelog [here](https://keepachangelog.com/en/1.0.0/)
 ### Added
 - Prepare to filter local SV frequency
 ### Changed
-- Speed up instituteS page loadiding by refactoring cases/institutes query
```
Comment (northwestwitch) [inline CHANGELOG.md:11]: 🤣

File: scout/adapter/mongo/case.py [hunk 2]
```diff
@@ -574,12 +574,16 @@ def nr_cases(self, institute_id=None):
         Returns:
```
Comment (northwestwitch) [inline scout/adapter/mongo/case.py:574]: I wonder if this query can be entirely skipped. One could sum up the numbers returned by [this other function](https://github.com/Clinical-Genomics/scout/blob/aecafa3219e3be066dbc137896d3d1b603dcd144/scout/server/blueprints/institutes/controllers.py#L420) (that is necessary) instead?

File: scout/constants/indexes.py [hunk 3]
```diff
@@ -123,8 +123,8 @@
         IndexModel([("synopsis", TEXT)], default_language="english", name="synopsis_text"),
         IndexModel([("causatives", ASCENDING)], name="causatives"),
         IndexModel(
-            [("collaborators", ASCENDING), ("updated_at", ASCENDING)],
-            name="collaborators_updated_at",
+            [("collaborators", ASCENDING), ("status", ASCENDING), ("updated_at", ASCENDING)],
```
Comment (northwestwitch) [inline scout/constants/indexes.py:126]: This speeds up the other query (nr_cases_by_status), no? Nice!

**Review by northwestwitch — APPROVED** [review]
It works. Check my comment about removing one query entirely!

File: scout/adapter/mongo/case.py (hunk 2 above)
Comment (dnil) [reply scout/adapter/mongo/case.py:574]: Yes, I thought the same. But it should be really quick, and then we have to ensure the others are not overlapping, and the function is not used elsewhere.

File: scout/constants/indexes.py (hunk 3 above)
Comment (dnil) [reply scout/constants/indexes.py:126]: Exactly!

File: scout/adapter/mongo/case.py (hunk 2 above)
Comment (dnil) [reply scout/adapter/mongo/case.py:574]: Ok, does not seem to be used elsewhere, and the `$group` should make it as non-overlapping as it gets.

File: scout/constants/indexes.py (hunk 3 above)
Comment (dnil) [reply scout/constants/indexes.py:126]: What is left to speed up if needed later is the third query, from `cases()` that actually cursors the documents.

File: scout/adapter/mongo/case.py (hunk 2 above)
Comment (northwestwitch) [reply scout/adapter/mongo/case.py:574]: Yes, I checked, it's used only in there

Comment (dnil) [reply scout/adapter/mongo/case.py]: Heh, it was surprisingly slightly slower when removing it; possibly my slower python juggling of dict-array conversions than the rather to the point db summing, or just a different day. But both still clocked faster than the original.

Screenshot 2023-04-18 at 10 44 10

I would still keep the latest without the double db request, just to keep things simple, and we see how it performs on a large db?

File: scout/server/blueprints/institutes/controllers.py [hunk 4]
```diff
@@ -416,8 +416,8 @@ def cases(store, request, institute_id):
     )
     all_cases = _sort_cases(data, request, all_cases)
 
-    data["nr_cases"] = store.nr_cases(institute_id=institute_id)
     data["status_ncases"] = store.nr_cases_by_status(institute_id=institute_id)
+    data["nr_cases"] = sum(data["status_ncases"].values())
```
Comment (northwestwitch) [inline scout/server/blueprints/institutes/controllers.py:420]: 👍🏻

File: scout/adapter/mongo/case.py (hunk 2 above)
Comment (northwestwitch) [reply scout/adapter/mongo/case.py]: Ok!

**Review by northwestwitch — APPROVED** [review]
👍🏻

**Pull request merged**